exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusMetricReaderTest.java

```diff
@@ -0,0 +1,1037 @@
+    // There is currently no API for adding scope attributes.
+    // However, we can at least test the otel_scope_version attribute.
+    Meter meter =
+        SdkMeterProvider.builder()
+            .setClock(testClock)
+            .registerMetricReader(this.reader)
+            .setResource(
+                Resource.getDefault().toBuilder().put("telemetry.sdk.version", "1.x.x").build())
+            .build()
+            .meterBuilder("test-scope")
+            .setInstrumentationVersion("a.b.c")
+            .build();
+    LongCounter counter = meter.counterBuilder("test.count").build();
+    counter.add(1);
+    String expected =
+        ""
+            + "# TYPE target info\n"
+            + "target_info{service_name=\"unknown_service:java\",telemetry_sdk_language=\"java\",telemetry_sdk_name=\"opentelemetry\",telemetry_sdk_version=\"1.x.x\"} 1\n"
+            + "# TYPE test_count counter\n"
+            + "test_count_total{otel_scope_name=\"test-scope\",otel_scope_version=\"a.b.c\"} 1.0\n"
+            + "test_count_created{otel_scope_name=\"test-scope\",otel_scope_version=\"a.b.c\"} "
+            + createdTimestamp
+            + "\n"
+            + "# EOF\n";
+    assertEquals(expected, toOpenMetrics(reader.collect()));
+  }
+
+  @Test
+  public void testOtelScopeDisabled() throws IOException {
+    PrometheusMetricReader reader = new PrometheusMetricReader(false);
+    Meter meter =
+        SdkMeterProvider.builder()
+            .setClock(testClock)
+            .registerMetricReader(reader)
+            .setResource(
+                Resource.getDefault().toBuilder().put("telemetry.sdk.version", "1.x.x").build())
+            .build()
+            .meterBuilder("test-scope")
+            .setInstrumentationVersion("a.b.c")
+            .build();
+    LongCounter counter = meter.counterBuilder("test.count").build();
+    counter.add(1);
+    String expected =
+        ""
+            + "# TYPE target info\n"
+            + "target_info{service_name=\"unknown_service:java\",telemetry_sdk_language=\"java\",telemetry_sdk_name=\"opentelemetry\",telemetry_sdk_version=\"1.x.x\"} 1\n"
+            + "# TYPE test_count counter\n"
+            + "test_count_total 1.0\n"
+            + "test_count_created "
+            + createdTimestamp
+            + "\n"
+            + "# EOF\n";
+    assertEquals(expected, toOpenMetrics(reader.collect()));
+  }
+
+  /**
+   * Unfortunately there is no easy way to use {@link TestClock} for Exemplar timestamps. The
+   * following is like {@code assertEquals()}, but {@code <timestamp>} matches arbitrary timestamps.
+   */
+  @SuppressWarnings("MethodCanBeStatic")
+  private void assertMatches(String expected, String actual) {
+    String[] parts = expected.split(Pattern.quote("<timestamp>"));
+    StringBuilder regex = new StringBuilder();
+    for (int i = 0; i < parts.length; i++) {
+      regex.append(Pattern.quote(parts[i]));
+      if (i <= parts.length - 2) {
```

Comment:
Why only if `i <= parts.length - 2`?

Reply:
Changed to a `stream()` based implementation which is easier to read.